Zend/zend_gc.c

```diff
@@ -365,13 +373,56 @@ ZEND_API void ZEND_FASTCALL gc_remove_from_buffer(zend_refcounted *ref)
 	}
 }
 
-static void gc_scan_black(zend_refcounted *ref)
+static gc_refcounted_stack* gc_refcounted_stack_init(int size)
+{
+	gc_refcounted_stack *stack;
+	stack = (gc_refcounted_stack*) emalloc(sizeof(gc_refcounted_stack));
```

Comment:
The gc_refcounted_stack structure has a fixed size, so we can stack allocate it. For that `stack_init()` should be changed to accept a pointer to the stack. You'd use it something like this:

    gc_refcounted_stack stack;
    gc_refcounted_stack_init(&stack, size);
    // Use stack
    gc_refcounted_stack_destroy(&stack);

Reply:
Understood, will change it to allocate on stack instead of heap.